extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -65,331 +70,376 @@ export const RTC_ERROR_ENUM = {
+            callout, errorReportingUrl, Date.now() - this.rtcStartTime_);
+    });
   }
-  let timeout;
-  try {
-    user().assert(rtcConfig['vendors'] || rtcConfig['urls'],
-        'RTC Config must specify vendors or urls');
-    Object.keys(rtcConfig).forEach(key => {
-      switch (key) {
-        case 'vendors':
-          user().assert(isObject(rtcConfig[key]), 'RTC invalid vendors');
-          break;
-        case 'urls':
-          user().assert(isArray(rtcConfig[key]), 'RTC invalid urls');
-          break;
-        case 'timeoutMillis':
-          timeout = parseInt(rtcConfig[key], 10);
-          if (isNaN(timeout)) {
-            user().warn(TAG, 'Invalid RTC timeout is NaN, ' +
-                        `using default timeout ${defaultTimeoutMillis}ms`);
-            timeout = undefined;
-          } else if (timeout >= defaultTimeoutMillis || timeout < 0) {
-            user().warn(TAG, `Invalid RTC timeout: ${timeout}ms, ` +
-                        `using default timeout ${defaultTimeoutMillis}ms`);
-            timeout = undefined;
-          }
-          break;
-        default:
-          user().warn(TAG, `Unknown RTC Config key: ${key}`);
-          break;
+
+  /**
+   * Attempts to parse the publisher-defined RTC config off the amp-ad
+   * element, then validates that the rtcConfig exists, and contains
+   * an entry for either vendor URLs, or publisher-defined URLs. If the
+   * config contains an entry for timeoutMillis, validates that it is a
+   * number, or converts to a number if number-like, otherwise overwrites
+   * with the default.
+   * IMPORTANT: If the rtcConfig is invalid, RTC is aborted, and the ad
+   *   request continues without RTC.
+   * @param {!Element} element
+   */
+  validateRtcConfig_(element) {
+    const defaultTimeoutMillis = 1000;
+    const unparsedRtcConfig = element.getAttribute('rtc-config');
+    if (!unparsedRtcConfig) {
+      return null;
+    }
+    const rtcConfig = tryParseJson(unparsedRtcConfig);
+    if (!rtcConfig) {
+      user().warn(TAG, 'Could not parse rtc-config attribute');
+      return null;
+    }
+    let timeout;
+    try {
+      user().assert(rtcConfig['vendors'] || rtcConfig['urls'],
+          'RTC Config must specify vendors or urls');
+      Object.keys(rtcConfig).forEach(key => {
+        switch (key) {
+          case 'vendors':
+            user().assert(isObject(rtcConfig[key]), 'RTC invalid vendors');
+            break;
+          case 'urls':
+            user().assert(isArray(rtcConfig[key]), 'RTC invalid urls');
+            break;
+          case 'timeoutMillis':
+            timeout = parseInt(rtcConfig[key], 10);
+            if (isNaN(timeout)) {
+              user().warn(TAG, 'Invalid RTC timeout is NaN, ' +
+                          `using default timeout ${defaultTimeoutMillis}ms`);
+              timeout = undefined;
+            } else if (timeout >= defaultTimeoutMillis || timeout < 0) {
```

Comment:
What does it meant to have a timeout of 0?  Should we even attempt to send the request?  Presumably even cache fetch takes some time.  I worry that "falsey" values like 0 could be misinterpreted by pubs